src/lookup-table/utils/dataTypeUtils.ts

```diff
@@ -0,0 +1,103 @@
+import { mapValues } from 'lodash'
+import { FieldMapType } from '../types'
+import { DateTime } from 'luxon'
+
+export const setDataTypes = (fieldMaps: FieldMapType[], rows: object[]) => {
+  // Determine type
+  const done: string[] = []
+  for (const row of rows) {
+    Object.entries(row).forEach(([key, value], index) => {
+      if (!done.includes(key)) {
+        const type = getType(value)
+        if (type !== null) {
+          fieldMaps[index].dataType = type
+          done.push(key)
+        }
+      }
+    })
+
+    if (done.length === fieldMaps.length) break
+  }
+
+  const fieldMap: { [key: string]: any } = fieldMaps.reduce((acc, curr) => {
+    const name = curr.fieldname
+    const type = curr.dataType
+    return { ...acc, [name]: type }
+  }, {})
+
+  // Now mutate rows based on those types
+  rows.forEach((row, index) => {
+    rows[index] = mapValues(row, (value, key) => {
+      if (key === 'id') return value
+      return convertType(value, fieldMap[key])
+    })
+  })
+}
+
+// Prepares table export by stringifying jsonb fields
+// Mutates rows *in-place*
+export const exportDataRows = (fieldMaps: FieldMapType[], rows: { [key: string]: any }[]) => {
+  const jsonFields = fieldMaps.filter((e) => e.dataType === 'jsonb').map((e) => e.fieldname)
+
+  rows.forEach((row) => {
+    jsonFields.forEach((jsonField: string) => {
+      row[jsonField] = row[jsonField] !== null ? JSON.stringify(row[jsonField]) : ''
+    })
+  })
```

Comment:
Without this, the json fields get exported as `[Object object]`, so this stringifies those fields. All other fields get dumped out as strings (like they were when imported).